src/plugins/data/server/index_patterns/routes/runtime_fields/create_runtime_field.ts

```diff
@@ -53,19 +54,18 @@ export const registerCreateRuntimeFieldRoute = (
         throw new Error(`Field [name = ${name}] already exists.`);
       }
 
-      indexPattern.addRuntimeField(name, runtimeField);
-
-      const addedField = indexPattern.fields.getByName(name);
-      if (!addedField) throw new Error(`Could not create a field [name = ${name}].`);
+      const createdFields = indexPattern.addRuntimeField(name, runtimeField as RuntimeField);
 
       await indexPatternsService.updateSavedObject(indexPattern);
 
-      const savedField = indexPattern.fields.getByName(name);
```

Comment:
Here too this validation is not necessary. We can wrap the `indexPatternsService.updateSavedObject()` in a `try/catch` but that operation should never remove the field created.